Mesa<=10.0.1 removed the GLXContextID from glx.h and it has been re-added since:

http://cgit.freedesktop.org/mesa/mesa/commit/?id=f425d56ba41382be04366d011536ee78a03a2f33

For now, use the same type (XID) than GLXContextID; we can remove this patch
once mesa-10 contains the above commit.

	-- xtraeme

--- a/addons/allegrogl/include/allegrogl/GLext/glx_ext_api.h	2014-01-05 09:08:26.796542194 +0100
+++ b/addons/allegrogl/include/allegrogl/GLext/glx_ext_api.h	2014-01-05 09:08:46.115682136 +0100
@@ -59,8 +59,8 @@ AGL_API(void, DestroyGLXVideoSourceSGIX,
 /* GLX_EXT_import_context */
 AGL_API(Display *, GetCurrentDisplayEXT, (void))
 AGL_API(int, QueryContextInfoEXT, (Display *, GLXContext, int, int *))
-AGL_API(GLXContextID, GetContextIDEXT, (const GLXContext))
-AGL_API(GLXContext, ImportContextEXT, (Display *, GLXContextID))
+AGL_API(XID, GetContextIDEXT, (const GLXContext))
+AGL_API(GLXContext, ImportContextEXT, (Display *, XID))
 AGL_API(void, FreeContextEXT, (Display *, GLXContext))
 #endif
 
